schema2Markdown.js

```diff
@@ -0,0 +1,260 @@
+        const typeString = properties?.type?.const;
+        const ref = root.$ref;
+                
+        markdownContent += `## Type\n\n`;
+        markdownContent += `\`${typeString}\`\n\n`;
+        markdownContent += `## Properties\n\n`; 
+
+        for (const [propertyName, propertyDetails] of Object.entries(properties)) {
+            if (propertyName != "type"){ 
+                markdownContent += processProperty(propertyName, propertyDetails, schema.examples);
+            }
+        }
+
+        if (ref) {
+            markdownContent += `ref: ${ref}\n\n`;
+        }
+
+        if (schema.examples && schema.examples.length > 0) {
+            if (schema.examples.length > 1) {
+                markdownContent += `## Examples\n\n`;
+            } else {
+                markdownContent += `## Example\n\n`;
+            }
+            
+            schema.examples.forEach((example) => {
+                markdownContent += '```json\n';
+                markdownContent += JSON.stringify(example, null, 2);
+                markdownContent += '\n```\n\n';
+            });
+        }
+
+        const frontMatter = generateFrontMatter(objectName, schema.description);
+
+        const outputFileName = `./website/versioned_docs/version-${version}/${schemaFolderName}/schemas/${title.replace(/\s+/g, '')}.md`;
+
+        fse.outputFileSync(outputFileName, `---\n${yaml.dump(frontMatter)}\n---\n\n${markdownContent}`);
+
+        // objectName must not contain any spaces
+        if (objectName != null) {
+            return schemaFolderName + '/schemas/' + objectName.replace(/\s+/g, '');
+        }
+    }
+}
+
+function escape(text) {
+    return text.replace(/>/g, '\\>').replace(/@experimental/g, '[@experiemental](/docs/fdc3-compliance#experimental-features)');
+}
+
+function generateFrontMatter(title, description) {
+    return {
+        title: title,
+        description: description,
+        sidebar_label: title,
+    };
+}
+
+function processSchemaFile(schemaFile, schemaFolderName, version) {
+    const schemaData = fse.readJSONSync(schemaFile);
+
+    // if there is allOf, then it is an object
+    const allOfArray = schemaData.allOf;
+    let sidebarItems = [];
+    if (Array.isArray(allOfArray) && allOfArray.length > 0) {
+        sidebarItems.push(generateObjectMD(schemaData, null, schemaFolderName, schemaFile, version));
+    }
+    if (schemaData.definitions) {
+        for (const [objectName, objectDetails] of Object.entries(schemaData.definitions)) {
+            sidebarItems.push(generateObjectMD(objectDetails, objectName, schemaFolderName, schemaFile, version));
+        }
+    }
+
+    return sidebarItems;
+}
+
+function parseSchemaFolder(schemaFolderName, version) {
+    // Read all files in the schema folder
+    const schemaFiles = fse.readdirSync("./schemas/"+schemaFolderName)
+    .filter(file => file.endsWith('.json'))
+    .map(file => path.join("./schemas/"+schemaFolderName, file));
```

Comment:
trips a semgrep rule as the first arg is passed to path.join a few lines below...  I can see that you don't actually take any user input so it either needs:
- a semgrep exception comment: https://semgrep.dev/docs/ignoring-files-folders-code
- don't take the folder as a parameter and just hardcode it to look at the context dirs (a comment at the top of the file that it only processes the context folders would be worth adding).